Memory_management_algorithm/best_fit.cpp

```diff
@@ -0,0 +1,133 @@
+#include <bits/stdc++.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+using namespace std;
+
+void best_fit(int memory_block[],int process_size[],int m,int n)
+{
+
+	int allocated_list[n][3];
+
+	int allocation_marking[m];
+
+	for(int i=0;i<m;i++)
+	 allocation_marking[i]=-1;
+
+	for(int i=0;i<n;i++)
+	{
+		int allocation=-1;
+		int mem_block=10000000;
+
+		for(int j=0;j<m;j++)
+		{
+			if((process_size[i]<=memory_block[j]) && (allocation_marking[j]== -1))
+			{
+				if(memory_block[j]<mem_block)
+				{
+					mem_block=memory_block[j];
+
+					allocation=j;
+				}
+			}
+
+		}
+		
+
+			if(allocation==-1)
+			{
+				allocated_list[i][0]=process_size[i];
+
+				allocated_list[i][1]=-1;
+
+				allocated_list[i][2]=-1;
+			}
+			else
+			{
+
+				allocated_list[i][0]=process_size[i];
+
+				allocated_list[i][1]=memory_block[allocation];
+
+				allocated_list[i][2]=allocation;
+
+				allocation_marking[allocation]=i;
+
+			}
+		
+	}
+
+	printf("Process Number     Process Size     Allocated memory block size     Memory block number\n");
+
+	for(int i=0;i<n;i++)
+	{
+		if(allocated_list[i][1]!=-1)
+		{
+			printf("%d                         %d                       %d                 %d\n",i,allocated_list[i][0],allocated_list[i][1],allocated_list[i][2]);
```

Comment:
Does not look good.

Reply:
I can't understand what do you mean by "does not look good" ? If you are pointing out the spaces in the **printf**, spaces are provided to format the output properly.

Reply:
Does not look good means these spaces are not looking good. You could have used tabs or something else..

Reply:
@SakshiShreya Changed the spaces with tabs